contracts/LaunchpegFactory.sol

```diff
@@ -0,0 +1,264 @@
+    /// @param _amountForDevs Amount of NFTs reserved for `projectOwner` (e.g 200)
+    /// @param _batchRevealSize Size of the batch reveal
+    /// @return launchpeg New Launchpeg address
+    function createLaunchpeg(
+        string memory _name,
+        string memory _symbol,
+        address _projectOwner,
+        address _royaltyReceiver,
+        uint256 _maxBatchSize,
+        uint256 _collectionSize,
+        uint256 _amountForAuction,
+        uint256 _amountForMintlist,
+        uint256 _amountForDevs,
+        uint256 _batchRevealSize
+    ) external override onlyOwner returns (address) {
+        address launchpeg = Clones.clone(launchpegImplementation);
+
+        isLaunchpeg[launchpeg] = true;
+        allLaunchpegs.push(launchpeg);
+
+        ILaunchpeg(launchpeg).initialize(
+            _name,
+            _symbol,
+            _projectOwner,
+            _royaltyReceiver,
+            _maxBatchSize,
+            _collectionSize,
+            _amountForAuction,
+            _amountForMintlist,
+            _amountForDevs,
+            _batchRevealSize
+        );
+
+        IBaseLaunchpeg(launchpeg).initializeJoeFee(
+            joeFeePercent,
+            joeFeeCollector
+        );
+
+        OwnableUpgradeable(launchpeg).transferOwnership(msg.sender);
+
+        emit LaunchpegCreated(
+            launchpeg,
+            _name,
+            _symbol,
+            _projectOwner,
+            _royaltyReceiver,
+            _maxBatchSize,
+            _collectionSize,
+            _amountForAuction,
+            _amountForMintlist,
+            _amountForDevs,
+            _batchRevealSize
+        );
+
+        return launchpeg;
+    }
+
+    /// @notice FlatLaunchpeg creation
+    /// @param _name ERC721 name
+    /// @param _symbol ERC721 symbol
+    /// @param _projectOwner The project owner
+    /// @param _royaltyReceiver Royalty fee collector
+    /// @param _maxBatchSize Max amount of NFTs that can be minted at once
+    /// @param _collectionSize The collection size (e.g 10000)
+    /// @param _amountForDevs Amount of NFTs reserved for `projectOwner` (e.g 200)
+    /// @param _batchRevealSize Size of the batch reveal
+    /// @param _salePrice Price of the public sale in Avax
+    /// @param _mintlistPrice Price of the whitelist sale in Avax
+    /// @return flatLaunchpeg New FlatLaunchpeg address
+    function createFlatLaunchpeg(
+        string memory _name,
+        string memory _symbol,
+        address _projectOwner,
+        address _royaltyReceiver,
+        uint256 _maxBatchSize,
+        uint256 _collectionSize,
+        uint256 _amountForDevs,
+        uint256 _batchRevealSize,
+        uint256 _salePrice,
+        uint256 _mintlistPrice
+    ) external override onlyOwner returns (address) {
+        address flatLaunchpeg = Clones.clone(flatLaunchpegImplementation);
+
+        isLaunchpeg[flatLaunchpeg] = true;
+        allLaunchpegs.push(flatLaunchpeg);
```

Comment:
I think that it should be better to add flatLaunchpeg to another array/mapping to differentiate them as they are different.

Reply:
As discussed with Patate, it might be better to do it like so if we add more launch types in the future

Reply:
As discussed with Patate, it might be better to do it like so if we add more launch types in the future

Reply:
Yes I agree

Reply:
Commit [90a2cf6](https://github.com/traderjoe-xyz/launchpeg/pull/11/commits/90a2cf6eae9d35cf9ec84ca26ed847b8b1c97275)